### PR TITLE
Amls 5082 -  Auth upgrades - Declaration flow

### PR DESCRIPTION
**Pull request by matthewharris1**
Added the new auth implementations to the declaration flow.

## How Has This Been Tested?
Unit Tested

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/controllers/declaration/AddPersonController.scala [hunk 1]
```diff
@@ -18,51 +18,49 @@ package controllers.declaration
...
 
 import scala.concurrent.Future
 
 class AddPersonController @Inject () (
                                        val dataCacheConnector: DataCacheConnector,
                                        val statusService: StatusService,
-                                       val authConnector: AuthConnector
-                                     ) extends BaseController {
+                                       authAction: AuthAction
+                                     ) extends DefaultBaseController {
```
Comment (slavvco) [inline app/controllers/declaration/AddPersonController.scala]: remove empty line

File: app/controllers/declaration/RegisterPartnersController.scala [hunk 2]
```diff
@@ -30,20 +29,37 @@ import models.status.{RenewalSubmitted, _}
...
 @Singleton
-class RegisterPartnersController @Inject()(val authConnector: AuthConnector,
+class RegisterPartnersController @Inject()(authAction: AuthAction,
                                            val dataCacheConnector: DataCacheConnector,
                                            implicit val statusService: StatusService,
                                            implicit val progressService: ProgressService
-                                          ) extends BaseController {
-
+                                          ) extends DefaultBaseController {
+  def businessPartnersView(amlsRegistrationNo: Option[String],
```
Comment (slavvco) [inline app/controllers/declaration/RegisterPartnersController.scala]: add line above to separate method from controller signature

File: app/controllers/declaration/WhoIsRegisteringController.scala [hunk 3]
```diff
@@ -18,7 +18,7 @@ package controllers.declaration
 
 import com.google.inject.Inject
 import connectors.{AmlsConnector, DataCacheConnector}
-import controllers.BaseController
+import controllers.{BaseController, DefaultBaseController}
 import forms.{EmptyForm, Form2, InvalidForm, ValidForm}
```
Comment (slavvco) [inline app/controllers/declaration/WhoIsRegisteringController.scala]: do you need to import BaseController?

File: app/controllers/declaration/RegisterResponsiblePersonController.scala [hunk 4]
```diff
@@ -17,27 +17,28 @@
 package controllers.declaration
 
 import javax.inject.{Inject, Singleton}
-
 import connectors.DataCacheConnector
-import controllers.BaseController
-import models.status.{SubmissionDecisionApproved, SubmissionReadyForReview, ReadyForRenewal}
+import controllers.{BaseController, DefaultBaseController}
```
Comment (slavvco) [inline app/controllers/declaration/RegisterResponsiblePersonController.scala]: do you need to import BaseController?

File: app/controllers/declaration/DeclarationController.scala [hunk 5]
```diff
@@ -18,7 +18,7 @@ package controllers.declaration
 
 import com.google.inject.Inject
 import connectors.DataCacheConnector
-import controllers.BaseController
+import controllers.{BaseController, DefaultBaseController}
```
Comment (slavvco) [inline app/controllers/declaration/DeclarationController.scala]: do you need to import BaseController?

File: app/controllers/declaration/RegisterPartnersController.scala (hunk 2 above)
Comment (matthewharris1) [reply app/controllers/declaration/RegisterPartnersController.scala]: Done

File: app/controllers/declaration/AddPersonController.scala (hunk 1 above)
Comment (matthewharris1) [reply app/controllers/declaration/AddPersonController.scala]: Done

File: app/controllers/declaration/RegisterPartnersController.scala (hunk 2 above)
Comment (matthewharris1) [reply app/controllers/declaration/RegisterPartnersController.scala]: Done

File: app/controllers/declaration/WhoIsRegisteringController.scala (hunk 3 above)
Comment (matthewharris1) [reply app/controllers/declaration/WhoIsRegisteringController.scala]: Done

File: app/controllers/declaration/RegisterResponsiblePersonController.scala (hunk 4 above)
Comment (matthewharris1) [reply app/controllers/declaration/RegisterResponsiblePersonController.scala]: Done

File: app/controllers/declaration/DeclarationController.scala (hunk 5 above)
Comment (matthewharris1) [reply app/controllers/declaration/DeclarationController.scala]: Done

**Pull request merged**